CMake generate step failing in `Linux_amd64 src-azurelinux-3.0-net9.0-crossdeps-graph` jobs

Several public PR builds are failing with the following error:

```
#8 5.342 CMake Error at /llvm-project.src/libcxx/cmake/Modules/HandleLibCXXABI.cmake:59 (add_library):
#8 5.342   No SOURCES given to target: libcxx-abi-headers-generate-private-headers
```

Below are three builds which have experienced the error:

- https://dev.azure.com/dnceng-public/public/_build/results?buildId=766292&view=logs&j=9451fe7a-3b19-5e32-5ae2-ab8409ac889d&t=8b49f318-ee4b-51db-c9b2-6c2579fdf8cb
- https://dev.azure.com/dnceng-public/public/_build/results?buildId=766733&view=logs&j=9451fe7a-3b19-5e32-5ae2-ab8409ac889d&t=8b49f318-ee4b-51db-c9b2-6c2579fdf8cb
- https://dev.azure.com/dnceng-public/public/_build/results?buildId=767687&view=logs&j=9451fe7a-3b19-5e32-5ae2-ab8409ac889d&t=8b49f318-ee4b-51db-c9b2-6c2579fdf8cb

I first noticed this in checks for [this PR](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1153), and the errors are unrelated to the changes in that PR.

--- a/src/azurelinux/3.0/net9.0/cross/riscv64-alpine/Dockerfile
+++ b/src/azurelinux/3.0/net9.0/cross/riscv64-alpine/Dockerfile
@@ -26,7 +26,7 @@ RUN TARGET_TRIPLE="riscv64-alpine-linux-musl" && \
         -DLIBCXX_ENABLE_SHARED=OFF \
         -DLIBCXX_HAS_MUSL_LIBC=ON \
         -DLIBCXX_CXX_ABI=libstdc++ \
-        -DLIBCXX_CXX_ABI_INCLUDE_PATHS="$ROOTFS_DIR/usr/include/c++/13.2.1/;$ROOTFS_DIR/usr/include/c++/13.2.1/$TARGET_TRIPLE" && \
+        -DLIBCXX_CXX_ABI_INCLUDE_PATHS="$ROOTFS_DIR/usr/include/c++/14.2.0/;$ROOTFS_DIR/usr/include/c++/14.2.0/$TARGET_TRIPLE" && \
     cmake --build runtimes -j $(getconf _NPROCESSORS_ONLN) && \
     cmake --install runtimes --prefix "$ROOTFS_DIR/usr"
 

